client/gutenberg/editor/controller.js

```diff
@@ -62,6 +65,48 @@ function waitForSiteIdAndSelectedEditor( context ) {
 	} );
 }
 
+function handleJetpackSSO( context ) {
+	// If we are dealing with an Atomic or Jetpack site we need to make sure that Jetpack SSO
+	// has been handled before we load the editor in an iFrame in order to prevent any issues
+	// with 3rd party cookie setting being blocked by the browser
+	const state = context.store.getState();
+	const currentRoute = getCurrentRoute( state );
+	const { URL: selectedSiteUrl, domain: selectedSiteDomain, jetpack } = getSelectedSite( state );
+
+	if ( ! jetpack ) {
+		return;
+	}
+
+	const {
+		hostname: parentDomain,
+		protocol: parentProtocol,
+		port,
+		query: parentQuery,
+	} = urlLib.parse( window.location.href, true );
+	const parentPort = port ? `:${ port }` : '';
+
+	// check query params from parent frame to check if we have already redirected back from Jetpack auth.
+	if ( parentQuery.jetpackSSO ) {
+		// If successfully redirected save to session storage so we don't need to redirect on every editor load
+		sessionStorage.setItem( `calypsoify_${ selectedSiteDomain }_jetpackSSO`, 'true' );
```

Comment:
`sessionStorage` will be cleared after you close the tab. If you open a new tab with the same URL, it will start with a clear session. Do you really want that? Is there a reason why you don't use `localStorage`?

Reply:
I think we need to stick with `sessionStorage` since we don't have any control over the 3rd-party cookies created by the custom domain when the current session finishes (i.e. the user could remove them after closing `wordpress.com/block-editor`).

Is there any other way to ensure that those cookies are always created before loading the editor?